### PR TITLE
Add arm64 support and correct the naming convention to the new standards

### PR DESCRIPTION
**Pull request by peterzhuamazon**
*Issue #, if available:*
V301153385

*Description of changes:*
This PR is to add arm64 support and correct the naming convention to the new standards

https://github.com/opendistro-for-elasticsearch/k-NN/actions/runs/473215772

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: jni/CMakeLists.txt [hunk 1]
```diff
@@ -81,7 +81,8 @@ set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "KNN JNI library built off of NMSLIB for O
 set(CPACK_PACKAGE_VENDOR "Amazon")
 set(CPACK_PACKAGE_CONTACT "Maintainer: ${KNN_MAINTAINER}")
 set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
-set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_PACKAGE_RELEASE}_${JVM_OS_TYPE}.${CMAKE_SYSTEM_PROCESSOR}")
+# set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_PACKAGE_RELEASE}_${JVM_OS_TYPE}.${CMAKE_SYSTEM_PROCESSOR}")
```
Comment (jmazanec15) [inline jni/CMakeLists.txt]: As opposed to commenting this out, can we just delete it?

File: .github/workflows/CD.yml [hunk 2]
```diff
@@ -5,13 +5,37 @@ on:
...
+          aws-secret-access-key: ${{ secrets.AWS_INFRATEST_EC2_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: AWS Cli Processing
+        run: |
+          #!/bin/bash
+          # Please add comma at the end of the RUNNERS strings
+          # Don't add space after commas
+          RUNNERS="odfe-knnlib-build-arm64"
+          export GIT_URL_REPO=${GITHUB_REPOSITORY}
+          ../opendistro-build/release-tools/scripts/setup_runners.sh run $RUNNERS ${{ secrets.ODFE_RELEASE_BOT_PUBLIC_PRIVATE_READ_WRITE_TOKEN }} ami-0ed17173ab64255b1
```
Comment (jmazanec15) [inline .github/workflows/CD.yml:30]: From my understanding, AMI Id's can change. How will we know when to change this?

File: .github/workflows/CD.yml [hunk 3]
```diff
@@ -103,3 +127,102 @@ jobs:
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      # Arm64 is not supported by setup-java actions as of now
+      # Placeholder in case it adds support later
+      # Using pre-installed jdk on arm64 self-hosted runners now
```
Comment (jmazanec15) [inline .github/workflows/CD.yml:154]: What version of Java will this be?

File: .github/workflows/CD.yml [hunk 4]
```diff
@@ -40,9 +64,9 @@ jobs:
           aws s3 cp $deb_artifact s3://artifacts.opendistroforelasticsearch.amazon.com/downloads/debs/opendistro-knn/
           aws cloudfront create-invalidation --distribution-id E1VG5HMIWI4SA2 --paths "/*"
 
-  library-build-and-ship-artifacts:
-    name: Build and release JNI library artifacts
-    runs-on: ubuntu-latest
+  library-build-and-ship-artifacts-x64:
```
Comment (jmazanec15) [inline .github/workflows/CD.yml:67]: Out of curiosity, why is it x64 instead of x86?

File: jni/CMakeLists.txt (hunk 1 above)
Comment (peterzhuamazon) [reply jni/CMakeLists.txt]: Will remove then, previously comment out for backing up.

File: .github/workflows/CD.yml (hunk 4 above)
Comment (peterzhuamazon) [reply .github/workflows/CD.yml:67]: This is based on the new naming convention, x86 is 32bit, x64 is 64bit

File: .github/workflows/CD.yml (hunk 2 above)
Comment (peterzhuamazon) [reply .github/workflows/CD.yml:30]: The only plugin that uses arm64 alongside infra release workflow is knnlib. All of the images are specifically made for github runners. If we change it on our side we will inform you guys that this value is changed.

File: .github/workflows/CD.yml (hunk 3 above)
Comment (peterzhuamazon) [reply .github/workflows/CD.yml:154]: As of now the runner is CentOS8 with JDK15.

**Review by jmazanec15 — APPROVED** [review]
LGTM! Thanks!

**Review by vamshin — APPROVED** [review]
LGTM!

**Pull request merged**